### PR TITLE
Verify multiple attestations

### PR DESCRIPTION
**Pull request by wilsonplau**
Attestor now accepts an array of AttestationTypes, checks allowlists of the specified attestationTypes and returns an array of signatures (or emails with a string representing multiple signatures).

### CODE REVIEW

**Comment by MixailE** [comment]
I tested the code and made sure that adding new methods doesn't break the old methods

**Review by MixailE — APPROVED** [review]

File: src/controllers/verify-ketl.ts [hunk 1]
```diff
@@ -1,30 +1,97 @@
...
+    for (const type of types) {
+      const allowlist = allowlistMap.get(type)
+      if (allowlist?.has(`token:${token}`))
+        attestations.push(signAttestationMessage(type, hexlifyString(token)))
+    }
+
+    if (!attestations.length)
+      return ctx.throw(
+        notFound(
+          `Couldn't find the invitation for this token. Please, try another one`
```
Comment (T-Damer) [inline src/controllers/verify-ketl.ts:43]: Let's use plain `""`

File: src/controllers/verify-ketl.ts [hunk 2]
```diff
@@ -1,30 +1,97 @@
...
+        )
+      )
+    return Promise.all(attestations)
```
Comment (T-Damer) [inline src/controllers/verify-ketl.ts:46]: You want to return a promise, that's right, we don't want to resolve it before return?

File: src/controllers/verify-ketl.ts [hunk 3]
```diff
@@ -1,30 +1,97 @@
...
+        const attestationHash = message[1]
+        secret.push(attestationHash)
+      }
+      secret.push(`t${type}${signature}`)
+    }
+
+    if (!secret.length)
+      return ctx.throw(
+        notFound(
+          `Couldn't find the invitation for this email address. Please, try another one`
```
Comment (T-Damer) [inline src/controllers/verify-ketl.ts:81]: Again, let's use `""`

File: src/helpers/getAllowlistMap.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import * as fs from 'fs'
+import { resolve } from 'path'
+import AttestationType from '@/models/AttestationType'
+
+function getAllowlist(attestationType: AttestationType) {
+  try {
+    const filePath = resolve(
+      process.cwd(),
+      'merkleTrees',
+      `${attestationType}.txt`
```
Comment (T-Damer) [inline src/helpers/getAllowlistMap.ts:10]: Sooner or later we need to remove `0, 1, 2...` into actual type names

Comment (T-Damer) [reply src/helpers/getAllowlistMap.ts:10]: If we'll have too much types (like 20), we might mess things up

File: src/helpers/getAllowlistMap.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    )
+    const file = fs.readFileSync(filePath, 'utf8')
+    const allowlist = file.split('\n')
+    const filteredAllowlist = allowlist.filter(
+      (record: string) => !/^#/.test(record) && record !== ''
+    )
+    return filteredAllowlist
+  } catch (e) {
+    return []
```
Comment (T-Damer) [inline src/helpers/getAllowlistMap.ts:19]: should we log it?

File: src/helpers/getAllowlistMap.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      process.cwd(),
+      'merkleTrees',
+      `${attestationType}.txt`
+    )
+    const file = fs.readFileSync(filePath, 'utf8')
+    const allowlist = file.split('\n')
+    const filteredAllowlist = allowlist.filter(
+      (record: string) => !/^#/.test(record) && record !== ''
+    )
+    return filteredAllowlist
```
Comment (T-Damer) [inline src/helpers/getAllowlistMap.ts:17]: Let's return without a const

File: src/controllers/verify-ketl.ts [hunk 7]
```diff
@@ -62,6 +161,70 @@ export default class VerifyKetlController {
...
+            hexlifyString(ownerAddress.toLowerCase()),
+            threshold,
+            hexlifyString(tokenAddress)
+          )
+        )
+    }
+    if (!attestations.length)
+      return ctx.throw(
+        notFound(
+          `Couldn't find the invitation for this Twitter handle. Please, try another one`
```
Comment (T-Damer) [inline src/controllers/verify-ketl.ts:222]: Here as well

File: src/controllers/verify-ketl.ts [hunk 8]
```diff
@@ -47,6 +114,38 @@ export default class VerifyKetlController {
...
+        if (allowlist?.has(`twitter:${id}`))
+          attestations.push(
+            signAttestationMessage(type, VerificationType.twitter, id)
+          )
+      }
+
+      if (!attestations.length)
+        return ctx.throw(
+          notFound(
+            `Couldn't find the invitation for this Twitter handle. Please, try another one`
```
Comment (T-Damer) [inline src/controllers/verify-ketl.ts:139]: Looks like we can extract this error

Comment (T-Damer) [reply src/controllers/verify-ketl.ts:139]: Like `handleInvitationError({ forThis: 'Twitter handle' })`

File: src/helpers/getAllowlistMap.ts (hunk 4 above)
Comment (MixailE) [reply src/helpers/getAllowlistMap.ts:10]: Yes, but we think about it later

**Comment by MixailE** [comment]
[Add multiple attestations for attestor/verify-ketl](https://trello.com/c/RRwj1OEy/3746-add-multiple-attestations-for-attestor-verify-ketl)

**Pull request merged**